src/main/java/org/mockito/internal/invocation/TypeSafeMatching.java

```diff
@@ -65,6 +65,6 @@ private static boolean isMatchesMethod(Method method) {
         if (method.isBridge()) {
             return false;
         }
-        return method.getName().equals("matches");
+        return "matches".equals(method.getName());
```

Comment:
Interesting feedback!

Personally, flip or no-flip has no difference to me for readability. String literal on LHS is in general safer so the change is good IMHO.